kickstarter/android-oss
# MBL-835: [Android] Update prelaunch, project, and rewards carousel screen_view event to use new mutation

# 📲 What

Modified existing and added new third party event mutation to prelaunch, project, and rewards carousel. 

# 🤔 Why

Adding support for GA4.

# 🛠 How

- Created a second use case that utilizes v2 in order to support prelaunch pages and the migration to rxjava2
- Updated capi mutation to third party mutation, and all corresponding tests
- Moved the event call from the rewards view holder view model to the rewards carousel to reflect when a user visits the reward carousel page screen

# 👀 See
Example of screen_view event triggered by the project page
![image](https://github.com/kickstarter/android-oss/assets/19390326/2f439d54-6fd9-4574-8dd1-7bfd4b166538)

# 📋 QA

- Make sure the consent management feature flag and either the ga4 and/or CAPI/Meta feature flags are enabled in remote config on the internal debug variant.
- Run app in internal debug variant
- Make sure you are logged in
- Find a project with sendThirdPartyEvent = true (i used the project "Be a Girl")
- Consent management must be enabled/allowed on your device (to trigger the dialog, either fresh install or clear all data)
- Monitor the call in android studio or find "thirdParty" in logcat
- Make sure the outgoing call includes the correct `firebaseScreen` and `firebasePreviousScreen` values (for example, if you search for the project "Be A Girl", when you visit the project page, the firebaseScreen would be "Project" and the firebasePreviousScreen would be "Search"
- We currently do not have any prelaunch projects that are accepting third party events, so i had to manually override the sendThirdPartyEvent value to true in Project.kt to trigger the event on a prelaunch project (search "Apex" and a prelaunch project should come up"

# Story 📖

[MBL-835: [Android] Update prelaunch, project, and rewards carousel screen_view event to use new mutation](https://kickstarter.atlassian.net/browse/MBL-835)
